src/components/SearchBox.js

```diff
@@ -0,0 +1,60 @@
+import React, { useRef, useEffect, useContext, useState } from 'react';
+import './SearchBox.css';
+import { LocationsContext } from '../context/LocationsContext';
+import { Icon } from '@iconify/react';
+import searchIcon from '@iconify-icons/bi/search';
+
+export default function SearchBox() {
+  const { coordinates, mapsApi, mapInstance, setNearByPlaces } = useContext(
+    LocationsContext,
+  );
+
+  const inputRef = useRef(null);
+
+  const [searchTerm, setSearchTerm] = useState('');
+  const [error, setError] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setSearchTerm(inputRef.current.value);
+  };
+
+  useEffect(() => {
+    if (searchTerm && mapsApi) {
+      const callback = (results, status) => {
+        if (status === 'ZERO_RESULTS') {
+          console.log('error');
+          setError(true);
+        }
+        if (status === mapsApi.places.PlacesServiceStatus.OK) {
+          setNearByPlaces(results);
+          setError(false);
+        }
+      };
+      const request = {
+        keyword: searchTerm,
+        location: coordinates,
+        radius: '500',
```

Comment:
Does this need to be a string?

Reply:
it's a string in the [google maps example](https://developers.google.com/maps/documentation/javascript/places?utm_source=google&utm_medium=cpc&utm_campaign=FY18-Q2-global-demandgen-paidsearchonnetworkhouseads-cs-maps_contactsal_saf&utm_content=text-ad-none-none-DEV_c-CRE_433476780397-ADGP_Hybrid%20%7C%20AW%20SEM%20%7C%20SKWS%20~%20Google%20Maps%20Places%20Library-KWID_43700039136946315-kwd-574199247741-userloc_9009670&utm_term=KW_%2Bplaces%20%2Blibrary-ST_%2Bplaces%20%2Blibrary&gclid=CjwKCAiAm-2BBhANEiwAe7eyFHtXkMhaN4tqdvtb9cbtzGiEK9_wT6VPTbeJ24-2KUa51MwgMxf8cBoCm2gQAvD_BwE#place_search_requests). We can try it as a number and see if it still functions the same.